src/js/components/CheckBoxGroup/__tests__/CheckBoxGroup-test.js

```diff
@@ -56,4 +71,89 @@ describe('CheckBoxGroup', () => {
     const tree = component.toJSON();
     expect(tree).toMatchSnapshot();
   });
+
+  test('onChange', () => {
+    const onChange = jest.fn();
+    const { container, getByText } = render(
+      <Grommet>
+        <CheckBoxGroup
+          options={[
+            { label: 'first-label', value: 'First' },
+            { label: 'second-label', value: 'Second' },
+          ]}
+          onChange={onChange}
+        />
+      </Grommet>,
+    );
+    fireEvent.click(getByText('first-label'));
+    expect(container.firstChild).toMatchSnapshot();
+  });
+
+  test('onClick', () => {
+    const { container, getByText } = render(
+      <Grommet>
+        <CheckBoxGroup
+          options={[
+            { label: 'first-label', value: 'First' },
+            { label: 'second-label', value: 'Second' },
+          ]}
+        />
+      </Grommet>,
+    );
+    fireEvent.click(getByText('first-label'));
+    expect(container.firstChild).toMatchSnapshot();
+    fireEvent.click(getByText('first-label'));
+    expect(container.firstChild).toMatchSnapshot();
```

Comment:
why does the same code happening twice? and how does onClick being tested (as implied on the name of the test)? I would expect an assertion for whether and how many times it had been called.

Reply:
It is happening twice because first it is clicking on the check box and marking it as checked, the second time it is unchecking the checkbox. I also added an assertion for onClick `expect(onClick).toBeCalledTimes(1);` at line 107 and 109.